### PR TITLE
ChIP-seq WF PE

### PR DESCRIPTION
**Pull request by lldelisle**
Mostly tested and wrapped by @nagoue

### CODE REVIEW

**Comment by lldelisle** [comment]
TODO:
Put more tests

**Comment by mvdbeek** [comment]
This one passed too!

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:9]:
```suggestion
        identifier: wt_H3K4me3
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:29]:
```suggestion
         wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:36]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:40]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:46]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:51]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:57]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/ChIPseq-tests.yml
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq-tests.yml:86]:
```suggestion
        wt_H3K4me3:
```

File: workflows/epigenetics/ChIPseq/README.md
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/README.md:10]:
```suggestion
- The peaks are called with macs2 which at the same time generates a coverage file
```
Which file format is this ? Might be nice to say
```
The peaks are called with macs2 which at the same time generates a coverage file in bedgraph format.
```
(no idea which format this is, just guessing here ...)

**Comment by lldelisle** [comment]
@nagoue Tu veux jeter un dernier coup d'oeil ? https://github.com/lldelisle/iwc/tree/finish_ChIPSeq/workflows/epigenetics/ChIPseq

**Comment by lldelisle** [comment]
For me it is ok now.
I wait for @nagoue to have a final look. @mvdbeek if you also want to check my last commits...

File: workflows/epigenetics/ChIPseq/ChIPseq.ga
Comment (mvdbeek) [inline workflows/epigenetics/ChIPseq/ChIPseq.ga:12]: I think we typically go with `MIT` for most Galaxy materials, any reason you prefer AGPL ?

Comment (lldelisle) [reply workflows/epigenetics/ChIPseq/ChIPseq.ga:12]: Not really, I can switch

**Comment by lldelisle** [comment]
Zenodo is slow. I think this is why CI is failing. I propose that we wait for the final review of @nagoue (end of the week) and then we try to make the CI pass again.

**Comment by mvdbeek** [comment]
This time that was me, I broke the hackathon branch of planemo, I'll rerun the tests once that's fixed.

**Comment by lldelisle** [comment]
No emergency. No need to rerun the tests before Friday.

**Comment by mvdbeek** [comment]
Alright, it passes tests again.

**Comment by lldelisle** [comment]
I feel like it would be better to call this workflow ChIP_PE as it uses PE and also I am interested in having the corresponding single read option. Is it ok for you?

**Comment by mvdbeek** [comment]
Yes, that's great! We can also have multiple workflows in a single directory, I think that's nice especially if there are some parts that are shared by the workflows.

**Comment by lldelisle** [comment]
Unfortunately the only step which are exactly the same are the 3 last (MACS2 report, MACS2 bigwig, multiQC). For all others there are parameters to adapt.
Should I put them in the same folder, or in different folder?

**Comment by mvdbeek** [comment]
> (MACS2 report, MACS2 bigwig, multiQC)

These could be a subworkflow, but I'm happy to improve this later. Maybe let's keep the SE/PE folders separate for now.

**Comment by lldelisle** [comment]
perfect

**Review by mvdbeek — APPROVED** [review]

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (lldelisle) [inline workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: Oups I think this is not working...

Comment (nagoue) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: May be explaining why I got an error at MACS2 step ?

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (lldelisle) [inline workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:29]:
```suggestion
          text: "<a href=\"#cutadapt_filtered_reads\" class=\"nav-l2\">Filtered Reads</a>"
```

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (lldelisle) [inline workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:31]:
```suggestion
          text: "<a href=\"#bowtie2\" class=\"nav-l1\">Bowtie 2 / HiSAT2</a>"
```

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (lldelisle) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: You got an error in planemo test?

Comment (nagoue) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: No, this is happening in running workflow on usegalaxy.org

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (mvdbeek) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:31]: You can just skip the outer quotes, they're not necessary:
```
In [8]: yaml.safe_load("""
   ...:         - that: "has_text"
   ...:           text: <a href="#bowtie2" class="nav-l1">Bowtie 2 / HiSAT2</a>
   ...: """)
Out[8]:
[{'that': 'has_text',
  'text': '<a href="#bowtie2" class="nav-l1">Bowtie 2 / HiSAT2</a>'}]
```

Comment (lldelisle) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:31]: thanks

File: workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml
Comment (lldelisle) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: This is not related then because this is just the test to run in the continuous integration on CI...
Can you share your history with me: lucille.delisle@epfl.ch

Comment (nagoue) [reply workflows/epigenetics/ChIPseq_PE/ChIPseq_PE-tests.yml:64]: I launch another run. I had an error with my genome size. I hope this is the reason why it failed otherwise I will share the history

**Comment by nagoue** [comment]
Tested on commit 5afdcbc: 
workflow run fine on Galaxy instance and planemo test went fine too.
congrats @lldelisle  !

**Comment by simleo** [comment]
> We can also have multiple workflows in a single directory

This is not really supported by [Workflow RO-Crate](https://about.workflowhub.eu/Workflow-RO-Crate/) (and thus WorkflowHub and LifeMonitor) if the workflows have "equal" status. One workflow has to be the *main workflow*, while other workflows in the crate can be e.g. subworkflows called by steps of the main workflow. What `gen_crates.py` is currently doing is simply to [pick the first entry](https://github.com/galaxyproject/iwc/blob/1af9ac8f37b1ed740198aa4af79cc7c89d0406f0/workflows/gen_crates.py#L229) from `.dockstore.yml`.

**Comment by lldelisle** [comment]
For me this workflow is ready

**Comment by mvdbeek** [comment]
Fantastic work @nagoue @lldelisle, this was a really productive EGD project!

**Comment by mvdbeek** [comment]
Good the know @simleo, no problem to keep them separate for now.

**Pull request merged**
